### PR TITLE
feat: [OI-204] vite migration

### PR DESCRIPTION
**Pull request by mmoio**
- Remove craco dependencies and configuration
- adds vite dependencies and configuration
- refactored envs to work with vite
   - Replaces REACT_APP with VITE as env prefixes
   - removes env-var and use vite import.meta

### CODE REVIEW

**Review by sebbalex — CHANGES_REQUESTED** [review]
Env file should respect the short environment names. This is mostly due to CIs which use this mapping to run trough envs.
So, I suggest to rename back `.env.production` to `.env.prod` and `.env.development` to `.env.dev`

**Comment by mmoio** [comment]
> Env file should respect the short environment names. This is mostly due to CIs which use this mapping to run trough envs. So, I suggest to rename back `.env.production` to `.env.prod` and `.env.development` to `.env.dev`

I'should ve explained this changes better in the pr description, but vite uses some defaults when loading env files, some documentation regarding this argument in this page https://vite.dev/guide/env-and-mode. One of this default behaviours is loading a .env.development in a dev environment, and .env.production when building.
I will need to make some changes regarding the uat environment as well creating a custom mode in the future,.

**Comment by mmoio** [comment]
> Env file should respect the short environment names. This is mostly due to CIs which use this mapping to run trough envs. So, I suggest to rename back `.env.production` to `.env.prod` and `.env.development` to `.env.dev`

For simplicity sake, I've changed back the environment files back to the original names, so to keep the pipeline environments coherent with the other parts of the project.
The only remaining change is renaming .env.local to .env.development, since is the one loaded by default by vite during development. A file .env.local is instead always loaded by vite and the defined variables in this file are overridden only if there is a more specific env file is loaded

**Comment by sebbalex** [comment]
> > Env file should respect the short environment names. This is mostly due to CIs which use this mapping to run trough envs. So, I suggest to rename back `.env.production` to `.env.prod` and `.env.development` to `.env.dev`
> 
> For simplicity sake, I've changed back the environment files back to the original names, so to keep the pipeline environments coherent with the other parts of the project. The only remaining change is renaming .env.local to .env.development, since is the one loaded by default by vite during development. A file .env.local is instead always loaded by vite and the defined variables in this file are overridden only if there is a more specific env file is loaded

Thanks for the clarification, now I have a more comprehensive view.
We can keep `.env.development` for local environment, since it is a local development file it doesn't need to have specific naming.
We should fix `.env.dev` because of its conflicts inside.
And eventually, we can think to have a `.env.local` file to gather all the common variables among different environment in it.

[Here you can find](https://github.com/pagopa/oneidentity/blob/1c66cc8a9cecd63aac83aef92681779835f8a86f/.github/workflows/deploy-fe-ci.yaml#L57) the deployment pipeline which will build and ship the frontend stuff.

This line will load the environment from matrix (which it can be one from "dev|uat|prod", line 17) and build the app:
`export $(grep -v ^# .env.${{ github.event.inputs.environment }} | xargs) && yarn build`

File: .github/workflows/deploy-fe-ci.yaml [hunk 1]
```diff
@@ -58,16 +58,16 @@ jobs:
           yarn lint
           yarn test:coverage
           if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
-              export $(grep -v ^# .env.${{ github.event.inputs.environment }} | xargs) && yarn build
+              yarn build --mode ${{ github.event.inputs.environment }}
           else
-              export $(grep -v ^# .env.dev | xargs) && yarn build
+              yarn build --mode
```
Comment (sebbalex) [inline .github/workflows/deploy-fe-ci.yaml:63]:
```suggestion
              yarn build --mode dev
```

File: .github/workflows/deploy-fe-ci.yaml [hunk 2]
```diff
@@ -77,24 +77,24 @@ jobs:
...
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502
         with:
           role-to-assume: ${{ vars.IAM_ROLE_DEPLOY_FE }}
           aws-region: ${{ matrix.region }}
-   
+
       - name: Copy to S3
         run: |
             aws s3 sync build s3://${{ vars.ASSETS_BUCKET_NAME }}
```
Comment (sebbalex) [inline .github/workflows/deploy-fe-ci.yaml:100]:
```suggestion
            aws s3 sync dist s3://${{ vars.ASSETS_BUCKET_NAME }}
```

**Review by sebbalex — APPROVED** [review]

**Pull request merged**